src/Sentry/SentryClient.cs

```diff
@@ -146,6 +146,15 @@ private SentryId DoSendEvent(SentryEvent @event, Scope? scope)
                 }
             }
 
+            // Each event should have an 'Environment' set. If none was set, then should default
+            // to 'production'.
+            if (string.IsNullOrWhiteSpace(@event.Environment))
+            {
+                @event.Environment = string.IsNullOrWhiteSpace(_options.Environment)
+                    ? "production"
+                    : _options.Environment;
```

Comment:
Environment should be pulled already from option elsewhere. So right now I assume we apply the value from options here and there. Could we have this logic there? It's probably on MainEventProcessor so the whole block could go there.

Reply:
When I didn't pull the options out (above), the ASP.NET Core tests (linked in the OP) were failing, though.

I wasn't sure if this was a brittle test -or- a great test that picked up my mistake ?

Reply:
Refactoring stuff.

Just found ` public SentryEvent Process(SentryEvent @event)` in `MainSentryEventProcessor`.